src/linreg.jl

```diff
@@ -18,9 +18,9 @@ systems than [`SolveQR`](@ref).
 """
 struct SolveCholesky <: AbstractLinregSolver end
 
-struct LinearRegressor{T<:Number}
+struct LinearRegressor{T<:Union{Matrix{<:Number},Vector{<:Number}}}
```

Comment:
Are you expecting to be able to handle Complex numbers? I would restrict the eltype to `<:Real` if not.